text/0000-feature-metadata.md

```diff
@@ -0,0 +1,309 @@
+-   `public`: A boolean flag defaulting to `true` that indicates whether or not
+    downstream crates should be allowed to use this feature
+-   `unstable`: A boolean flag indicating that the feature is only usable with
+    nightly Rust, or is otherwise not API-stable
+
+If a downstream crate attempts to use the features `baz` and `qux`, they will
+see messages like the following:
+
+```
+warning: feature `quux` on crate `mycrate` is deprecated since version
+  1.2.3: "don't use this!"
+
+error: feature `baz` on crate `mycrate` is private and cannot be used by
+  downstream crates
+```
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+Validation and parsing of the new schema, described above, shoud be relatively
+straightforward. Each of the added keys is discussed individually in the
+following sections:
+
+## `requires`
+
+`requires` will take the place of the feature dependency array that currently
+exists. Semantics will remain unchanged.
+
+This is a required key.
+
+## `doc`
+
+`doc` is the most straightforward: it accepts markdown-flavored text, and should
+be thought of as the equivalent to a `#[doc(...)]` attribute. Like doc comments,
+the first line should be usable as a summary, though most use cases will only
+have one line.
+
+There is nothing in this RFC that cargo `must` do with this action, since it is
+mainly intended for the consumption of `rustdoc` or `docs.rs`. However, it can
+be used for general diagnostic information.
+
+## `deprecated`
+
+`deprecated` should be thought of as the equivalent of the [`deprecated`]
+attribute in Rust source. The value can be a boolean, string, or an object with
+`since` or `note` keys. Schema rules are as follows:
+
+-   If a boolean value, `false` indicates not deprecated and `true` indicates
+    deprecated
+-   If an object, the keys `since` and/or `note` can be specified
+-   An empty object is not allowed to avoid the ambiguous
+    `foo = { deprecated = {} }`
+-   A string `foo = { deprecated = "my msg" }` will be equivalent to if that string
+    was specified in the `note` field:
+    `foo = { deprecated = { note = "my msg" } }`
+-   If not specified, the default is `false`
+
+If a downstream crate attempts to use a feature marked `deprecated`, Cargo
+should produce a warning. There are two exceptions to this:
+
+-   If a `since` is specified that is later than the crate's current version,
+    this warning should not be emitted.
+-   This warning should not be emitted for crates that reexport the feature
+    under a feature also marked deprecated. For example: crate `foo` exports
+    feature `phooey`, and crate `bar` exports feature `barred = ["foo/phooey"]`.
+    If `foo` markes `bar` as deprecated, checking `bar` will emit a warning
+    unless `barred` is also marked `deprecated.
+
+## `public`
+
+`public` is a boolean value that defaults to `true`. It can be thought of as
```

Comment:
As an update, we are talking about public/private dependencies going back to the [RFC stage](https://github.com/rust-lang/rust/issues/44663#issuecomment-1572992175).  Maybe its hidden in the comments but I've not seen a reason for `public` vs `pub` so we *might* want to reconsider that.  I can ask around.

Reply:
I think I like `public` as a key name a bit better than `pub`  since it seems like we tend not to shorten words in key values or in attributes (`doc` being a notable exception). Consistency is a good point though, so I will update it to `pub` if you would prefer.

Reply:
I think leaving it as-is is fine but we should note that we are being consistent with the approved-but-not-implemented public as an alternative to being consistent with rust syntax.

Reply:
Added the suggested comments